app/controller/AppController.php

```diff
@@ -65,25 +65,55 @@ public function accumulate(string $value)
     {
         try {
             $this->manager->append($value);
-            $chaine = $this->manager->getAccumulator();
-            $this->index($chaine);
-        } catch(Exception $e) {
+            $this->index();
+        } catch (Exception $e) {
             $this->error($e->getMessage());
         }
     }
 
     /**
      * Lance une action (opération, égal, pourcentage, ...)
      * @param string $action
+     * @throws Exception
      */
     public function action(string $action)
     {
-        switch($action) {
-            case 'clear';
+        $result = '';
+        $input = '';
+        switch ($action) {
+            case 'clear':
                 $this->manager->reset();
+                $result = $this->manager->getResult();
+                break;
+            case 'plus':
+                $this->manager->operator(Calculator::PLUS);
+                $result = $this->manager->getResult();
+                $input = $this->manager->getInput();
+                break;
+            case 'minus':
+                $this->manager->operator(Calculator::MINUS);
+                $result = $this->manager->getResult();
+                $input = $this->manager->getInput();
+                break;
+            case 'times':
+                $this->manager->operator(Calculator::TIMES);
+                $result = $this->manager->getResult();
+                $input = $this->manager->getInput();
                 break;
-            default;
+            case 'divide':
+                $this->manager->operator(Calculator::DIVIDE);
+                $result = $this->manager->getResult();
+                $input = $this->manager->getInput();
+                break;
+            case 'equals':
+                $this->manager->calculate();
+                $result = $this->manager->getResult();
+                $input =$this->manager->getInput();
+                break;
+            default:
+                throw new Exception('La fonction n\'est pas encore implémentée');
         }
-        $this->index();
+        $this->render($result, $input);
```

Comment:
Je pense que c'est pour ça que tu avais un bug sur le clear().
Tu as remplacé `$this->index();` par `$this->render($result, $input);` qui fait exactement la même chose.